web/src/routes/Layout/index.tsx

```diff
@@ -183,12 +192,36 @@ const PublicRoute: React.FunctionComponent<any> = ({
     );
   }
 
-  const renderCustomerComponent = (props: RouteComponentProps) => <CustomComponent {...props} />;
+  const renderCustomerComponent = (props: RouteComponentProps) =>
+    CustomComponent ? <CustomComponent {...props} /> : null;
+
+  return <Route {...rest} render={render || renderCustomerComponent} />;
+};
+
+const LangHandler: FC = ({ children }) => {
+  const dispatch = useDispatch();
+  const { lang } = useParams<{ lang: string }>();
```

Comment:
```suggestion
  const { lang } = useParams<{ lang?: string | undefined }>();
```
Думаю так тип будет точнее

Reply:
на undefined ругается generic из useParams, поэтому так
```
const { lang } = useParams<{ lang?: string }>();
```